src/Client/OfficeInterop/OfficeInterop.fs

```diff
@@ -1109,12 +1116,107 @@ let addBuildingBlock (excelTable:Table) (arcTable:ArcTable) (newBB:CompositeColu
+
+            let excelTableName =
+                if excelTableName.IsSome then excelTableName.Value
+                else failwith "No excel table name has been found!"
+
+            //When a name is available get the annotation and arctable for easy access of indices and value adaption
+            //Annotation table enables a easy way to adapt the table, updating existing and adding new columns
+            let! excelTable = tryGetAnnotationTableByName context excelTableName
+
+            //Arctable enables a fast check for the existence of input- and output-columns and their indices
+            let! arcTable =                
+                if excelTable.IsSome then
+                    ArcTable.tryGetFromExcelTable(excelTable.Value, context)
+                else failwith "No excel table has been found!"
+
+            //When both tables could be accessed succesfully then check what kind of column shall be added an whether it is already there or not
+            if arcTable.IsSome then
+
+                arcTable.Value.Join(tableToAdd,?index=index, ?joinOptions=options, forceReplace=true)
+
+                let tableValues = arcTable.Value.ToExcelValues() |> Array.ofSeq                
+                let (headers, body) = Array.ofSeq(tableValues.[0]), ResizeArray[tableValues.[1]]                
+
+                let newTableRange = excelTable.Value.getRange()
+                
+                let _ = newTableRange.load(propertyNames = U2.Case2 (ResizeArray["rowIndex"; "columnIndex"; "columnCount"; "rowCount"; "address"; "isEntireColumn"; "worksheet"; "values"]))
+
+                do! context.sync().``then``(fun _ ->
+                    excelTable.Value.delete()
+                )
+
+                let! (newTable, _) = createAnnotationTableAtRange(false, false, newTableRange, context)
+
+                let _ = newTable.load(propertyNames = U2.Case2 (ResizeArray["name"; "values"; "columns"]))
+
+                do! context.sync().``then``(fun _ ->
+
+                    newTable.name <- excelTableName
+
+                    let headerNames =
+                        let names = headers |> Array.map (fun item -> item.Value.ToString())
+                        names
+                        |> Array.map (fun name -> Indexing.extendName names name)
+
+                    headerNames
+                    |> Array.iteri(fun i header ->                        
+                        ExcelHelper.addColumn i newTable header (arcTable.Value.RowCount + 1) "" |> ignore)
+                )
+
+                let bodyRange = newTable.getDataBodyRange()
+
+                let _ = bodyRange.load(propertyNames = U2.Case2 (ResizeArray["columnCount"]))
+
+
+                do! context.sync().``then``(fun _ ->
+
+                    //We delete the annotation table because we cannot overwrite an existing one
+                    //As a result we create a new annotation table that has one column
```

Comment:
The function createAnnotationTableAtRange uses only the start position of the given range because we do not reuse existing data because that data is already part of the arctable